lib/smart_proxy_openscap/arf_parser.rb

```diff
@@ -6,12 +6,41 @@ module OpenSCAP
     class ArfParser
       include Proxy::Log
 
+      EXCEPTION_MAP = {
+        Proxy::OpenSCAP::ReportDecompressError => 1,
+        Nokogiri::XML::SyntaxError => 2
+      }
+
       def initialize(cname, policy_id, date)
         @cname = cname
         @policy_id = policy_id
         @date = date
       end
 
+      def as_json_forked(arf_data)
+        r, w = IO.pipe
+        if child_id = Process.fork
+          w.close
+          data = r.read
+          Process.wait(child_id)
```

Comment:
```suggestion
          data = r.read
          r.close
          Process.wait(child_id)
```
I guess we need to close that as well.